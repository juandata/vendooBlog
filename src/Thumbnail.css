.ArticleWrapper {
    display: flex;
    align-items: flex-end;
    background-size: 100%;
    min-height: 248px;
    background-position: bottom;
    margin-bottom: 11px;
    transition: 0.3s;
 
}
.ArticleWrapper:hover {
    background-size: 110%;
}
.SmallThumbnail {
    min-height: 100px;
    flex-grow: 1;
}
.PicturesAllInfoWrapper {
    display: flex;
    width: 100%;
    display: flex;
    color : #ffffff;
    flex-direction: column;
}   

.ArticleHeader {
    margin-left: 20px;
    width: 290px;
height: 39px;
font-family: Montserrat;
font-size: 28px;
font-weight: 600;

color: #ffffff;
text-decoration: none solid rgb(255, 255, 255);
}
.PicturesInfoWrapper {
    display: flex;
    justify-content: space-between;
}
.LikesInfo {
   margin-left: 20px;
   width: 62px;
height: 18px;
font-family: Montserrat;
font-size: 14px;
color: #ffffff;
text-decoration: none solid rgb(255, 255, 255);
letter-spacing: 1px;
font-weight: 400;
}

.DateInfo {
    margin-right: 20px;
    width: 63px;
height: 18px;
font-family: Montserrat;
font-size: 14px;
color: #ffffff;
text-decoration: none solid rgb(255, 255, 255);
letter-spacing: 1px;
text-align: right;
font-weight: 400;
}
.SmallFont {
    width: 290px;
height: 22px;
font-family: Montserrat;
font-size: 18px;
color: #ffffff;
text-decoration: none solid rgb(255, 255, 255);
}
.ThumbnailSection2 {
    min-height: 380px;
}