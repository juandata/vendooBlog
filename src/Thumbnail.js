import React from "react";
import "./Thumbnail.css";

const Thumbnail = props => {
  const date = props.Date;
  const id = props.Id;
  const className =  date === undefined ? 'ArticleWrapper SmallThumbnail' : id === undefined ?'ArticleWrapper' :
  'ArticleWrapper ' + id ;
  const titleClassName = date === undefined ? "ArticleHeader SmallFont" : 'ArticleHeader';
  return(
    <article className={className} style={{backgroundImage: `url(${props.Img})`
  }}>
    <div className="PicturesAllInfoWrapper">
    <header className={titleClassName}>{props.Title}</header>
      <div className="PicturesInfoWrapper" > 
          
      <p className="LikesInfo">{props.Likes} Likes</p>
      {date === undefined ? null : <p className="DateInfo">{props.Date}</p>}
      
      </div>
    </div>
      
    </article>
  )
 
}
     
 


export default Thumbnail;

