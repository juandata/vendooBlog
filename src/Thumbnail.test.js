import React from 'react';
import { shallow } from 'enzyme';
import Thumbnail from './Thumbnail';
import snowMountainImg from './assets/snowMountains.png';


it('Thumbnail should render correctly with no props', () => {
    const component = shallow(<Thumbnail/>);
    expect(component).toMatchSnapshot();
  });

  it('Thumbnail mouseover should make image effect with given props', () => {
    const component = shallow(
    <Thumbnail Title="Snow Mountains" Likes={10} Date={"12/18/90"} Img={snowMountainImg} />
    );
    component
      .simulate('mouseenter');
    expect(component).toMatchSnapshot();
    component.unmount();
  });