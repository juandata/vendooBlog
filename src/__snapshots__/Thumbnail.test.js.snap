// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail mouseover should make image effect with given props 1`] = `
Object {
  "print": [Function],
  "test": [Function],
}
`;
